### PR TITLE
Integrate PARQ with lowbit Arm CPU kernels

### PR DESCRIPTION
**Pull request by metascroy**
This PR integrates PARQ with lowbit CPU kernels by translating PARQ to an LUT tensor.

### CODE REVIEW

File: test/prototype/test_dynamic_activation_lut.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    quantize_(parq_model, config)
+
+    # Apply dynamic activation to parq model.  This will serve as the LUT reference
+    parq_model_with_dyn_quant = deepcopy(parq_model)
+    quantize_(
+        parq_model_with_dyn_quant,
+        Int8DynamicActivationConfig(),
+        # We have to explicitly provide filter_fn because the default linear filter
+        # excludes modules with AffinQUnatizedTensor weights
+        filter_fn=lambda m, fqn: isinstance(m, torch.nn.Linear),
```
Comment (lisjin) [inline test/prototype/test_dynamic_activation_lut.py:104]: Should I add a patch somewhere so that StretchedAffineQuantizedTensor is included in the default filter_fn?

File: test/prototype/test_dynamic_activation_lut.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        )
+    )
+    quantize_(lut_model, conversion_config, filter_fn=conversion_config.get_filter_fn())
+
+    # Run both models and compare
+    parq_out = parq_model(activations)
+    parq_with_dyn_quant_out = parq_model_with_dyn_quant(activations)
+    lut_out = lut_model(activations)
+
+    assert torch.allclose(parq_out, parq_with_dyn_quant_out, atol=1e-1, rtol=1e-1)
```
Comment (lisjin) [inline test/prototype/test_dynamic_activation_lut.py:121]: Just curious, is using such a high atol and rtol expected when comparing to the model with 8-bit activations? Maybe this means we should always use quantized activations during QAT in the future.

File: torchao/prototype/quantization/dynamic_activation_lut/api.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    # Construct LUT as 2 * ([q_min, q_max] - 0.5)
+    assert torch.all(zero_point == -0.5)
+    lut = torch.arange(q_min, q_max + 1)
+    lut = 2 * lut + 1
+
+    # Construct idx values
+    qval_idx = int_data - q_min
+
+    # Construct scale
+    scale = scale.reshape(-1).to(torch.float32)
```
Comment (lisjin) [inline torchao/prototype/quantization/dynamic_activation_lut/api.py:71]: Should this be using weight.dtype instead of torch.float32? This is what we use in [choose_qparams_stretched_affine](https://github.com/pytorch/ao/blob/3515cb66b6b0a8f538d302877a81eb52632eee20/torchao/prototype/parq/quant/quant_api.py#L39), which follows [_choose_qparams_affine](https://github.com/pytorch/ao/blob/main/torchao/quantization/quant_primitives.py#L1533).

**Review by lisjin — APPROVED** [review]
This looks great to me—thanks @metascroy!

File: test/prototype/test_dynamic_activation_lut.py (hunk 2 above)
Comment (metascroy) [reply test/prototype/test_dynamic_activation_lut.py:121]: I think most entries are accurate within 2 decimal places, but for CI it's important to give a buffer because we don't want it to be flakey.

But we will know more on dynamic quant accuracy evaluations from the work you're doing now

File: torchao/prototype/quantization/dynamic_activation_lut/api.py (hunk 3 above)
Comment (metascroy) [reply torchao/prototype/quantization/dynamic_activation_lut/api.py:71]: The kernel only supports fp32 scales, so we cast to fp32, no matter the original dtype.

File: test/prototype/test_dynamic_activation_lut.py (hunk 1 above)
Comment (metascroy) [reply test/prototype/test_dynamic_activation_lut.py:104]: Let's hold off on that now.

I'd like to get thoughts from people in ao first
